addons/mail/models/mail_thread.py

```diff
@@ -1646,6 +1646,17 @@ def _message_partner_info_from_emails(self, emails, link_mail=False):
     # MESSAGE POST API
     # ------------------------------------------------------
 
+    def _message_post_check_attachments(self, attachment_ids, attachment_tokens):
```

Comment:
Given `mail.thread` is a mixin, I don't know that this API is sensible when it can (and should?) not be overridden on a per-model basis.

Reply:
It has effectively nothing to do with standard mail.thread code ... just internals related to attachments. I don't think we really want to enable model-based management, even with a static management of attachments it is already a complete mess ^^ .

Reply:
I think either a free function (to import normally) or on mail.message would make sense.

Alternatively, a sibling function of `_document_check_access`: in essence it's just a way to easily mass-call that, so either put them next to one another, or just have `_document_check_access` work on multiple records and tokens directly.

Reply:
Ok that looks like a good idea to only have one function, I guess I'll just copy the code from portal and adapt it to work specifically in batch for attachments, rather than trying to re-use it maybe incorrectly.

Reply:
Does not seem message model has any link with this, why would we put that on message model ? But if this can be done generic (on BaseModel ?), better indeed.